src/System.Buffers.Primitives/System/Buffers/Pooling/MemoryPool.cs

```diff
@@ -5,9 +5,13 @@ namespace System.Buffers
 {
     public abstract class MemoryPool<T> : IDisposable
     {
+        public const int AnySize = int.MinValue;
+
         public static MemoryPool<T> Default => Internal.ArrayMemoryPool<T>.Shared;
 
-        public abstract OwnedMemory<T> Rent(int minimumBufferSize);
+        public abstract OwnedMemory<T> Rent(int minimumBufferSize = AnySize);
+
+        public abstract int MaxBufferSize { get;  }
```

Comment:
Why is this required?

Reply:
So callers can query it to know that Rent won't throw (like your slab pool does)

Reply:
How would they react to the knowledge?

Reply:
Usually renting code lives deep inside the library and won't be able to do anything except to throw, but pool already does throwing.

Reply:
I don't agree with "and won't be able to do anything" part. 

Also, see https://github.com/dotnet/corefxlab/blob/010238eb489ccfb04a47774cc606653afb82464a/src/System.IO.Pipelines.Networking.Sockets/SocketConnection.cs#L356

Reply:
Okay, I'm fine with keeping it, maybe consumer would be able to adjust to block size.

Reply:
What if there is no limit? `int.MaxValue`? Maybe it's better to make it nullable to make it more explicit?

Reply:
int.Max value is the limit, as OwnedMemory<T> cannot be larger.